packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -81,4 +82,52 @@ export default class Pools {
       await _xo.mergeXenPools(source._xapiId, target._xapiId, force)
     }
   }
+
+  async listPoolsMatchingCriteria({
+    hostVersion,
+    minHostCpus = 0,
+    minHostMemory = 0,
+    minPoolCpus = 0,
+    minSrSize = 0,
+    poolName,
+    srName,
+  }) {
+    const hostsByPool = {}
+    const srsByPool = {}
+    const pools = []
+    for (const obj of this._xo._objects.values()) {
+      if (obj.type === 'host') {
+        if (hostsByPool[obj.$pool] === undefined) {
+          hostsByPool[obj.$pool] = []
+        }
+        hostsByPool[obj.$pool].push(obj)
+      } else if (obj.type === 'SR') {
+        if (srsByPool[obj.$pool] === undefined) {
+          srsByPool[obj.$pool] = []
+        }
+        srsByPool[obj.$pool].push(obj)
+      } else if (obj.type === 'pool') {
+        pools.push(obj)
+      } else {
+        continue
+      }
+    }
+
+    return pools.filter(
+      pool =>
+        (poolName === undefined || new RegExp(poolName).test(pool.name_label)) &&
+        pool.cpus.cores >= minPoolCpus &&
+        some(
+          hostsByPool[pool.id],
+          host =>
+            (hostVersion === undefined || versionSatisfies(host.version, `=${hostVersion}`)) &&
+            host.cpus.cores >= minHostCpus &&
+            host.memory.size >= minHostMemory
```

Comment:
Is it relevant to match against the whole memory or do users prefer matching against available memory?